### ISSUE TITLE
Query regarding how to distribute raw data

### ISSUE THREAD

**Issue by atikaakmal** [joeynmt/joeynmt#120]
Hi,

I am trying to run joeynmt model by this config file **(transformer_iwslt14_deen_bpe.yaml)**. I ran sucessfull but the blue score is very low like 23.18. I supposed that might be my dataset is small thats why I am getting low blue score. I have  few questions regarding this. I am very new user and dont have much experience.

1. what should be the size of train, valid and test file?.
2. I already have much rawdata for train file but from where i can download data for valid.
3. should I use same parameter that are give by joenmt in a config file or can i change parameter like **batch_size, logging_freq, validation_freq, num_layers and num_heads**.
4. is it above parameters are according to the size of data. 

Also, currently the size of data that I am using is following after tokenization and bpe fomatting:
train: 769107 KB
valid: 120,705 kB
test: 441 KB

Thanks

**Comment by juliakreutzer** [comment]
Hi @atikaakmal,

I need a bit more information to understand the context of the problem/BLEU score: 
1. Which language pair are you working on? 
2. Which domain does the data come from? 
3. How many parallel sentences are in each of your sets?
In general it is hard to generally classify BLEU scores as high or low, since they depend on the conditions of the problem. Do you have any benchmark score to compare yours against? Did you take a look at the outputs, are they of any use?

Regarding your questions:
1. train: as big as possible (as long as it's reasonably clean), valid/test: usually around 2-3k sentences - just make sure they're not overlapping :) 
2. Depends on the language pair and domain, your valid set should reflect the task you have in mind (e.g. translating legal texts). If you haven't, check out OPUS (http://opus.nlpl.eu/) to find data. You might also take a part of your training data to use as validation set (remove it from training).
3. Good idea to change them as well, as soon as you have a working model (23.18 BLEU seem valid), you can play around with these to see how they affect the performance. To have a fair comparison, make sure to increase the logging_freq and validation_freq as you decrease batch_size or vice versa, since these are counted in number of batches not overall training samples.
4. If your data is small, you might have higher chances with a smaller, more strongly regularized model (e.g. decreasing number of layers, increasing label smoothing or dropout). Here's a paper that investigates this: https://arxiv.org/abs/2011.02266.

Hope that helps :)

**Comment by juliakreutzer** [comment]
Closing due to inactivity. @atikaakmal please re-open if you still have problems with the data.
